### PR TITLE
Add support of multiple APNS configs

### PR DESCRIPTION
**Pull request by SusanDoggie**
Allow to swap from difference configs

### CODE REVIEW

File: Sources/APNS/Application+APNS.swift [hunk 1]
```diff
@@ -1,49 +1,70 @@
...
+                    self.configurations.use(config, as: "default", isDefault: true)
+                }
             }
         }
 
         struct PoolKey: StorageKey, LockKey {
             typealias Value = EventLoopGroupConnectionPool<APNSConnectionSource>
         }
 
-        public var pool: EventLoopGroupConnectionPool<APNSConnectionSource> {
```
[the remark this answers is not in the document]
Comment (SusanDoggie) [reply Sources/APNS/Application+APNS.swift:26]: Or just putting all the new stuff under a subscript method?

Maybe like this:

    let pool = app.apns["test-push"].pool

[the remark this answers is not in the document]
Comment (SusanDoggie) [reply Sources/APNS/Application+APNS.swift:26]: I add the pool property back.

File: Sources/APNS/Application+APNS.swift [hunk 2]
```diff
@@ -1,49 +1,75 @@
...
-                self.application.storage.set(PoolKey.self, to: new) {
-                    $0.shutdown()
-                }
-                return new
-            }
+            return self.configurations.pool(logger: self.application.logger,
+                                            on: self.application.eventLoopGroup)
+        }
+        
+        public func pool(for configID: Application.APNS.ConfigID? = nil) -> EventLoopGroupConnectionPool<APNSConnectionSource> {
```
Comment (SusanDoggie) [inline Sources/APNS/Application+APNS.swift:68]: Maybe this method should be changed with a good name.

File: Sources/APNS/Application+APNS.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (kylebrowning) [reply Sources/APNS/Application+APNS.swift:26]: Agreed. Id also be okay with a new major version if its too much hassle, this is a welcome change!

[the remark this answers is not in the document]
Comment (kylebrowning) [reply Sources/APNS/Application+APNS.swift:26]: im picking this up again since ill be doing a major change for APNSwift

**Comment by kylebrowning** [comment]
Just another update, I should have something very soon, likely before the weekend.

**Comment by kylebrowning** [comment]
okay Ill need to refactor this PR to get it working correct, but #46 is ready to lay the groundwork

**Comment by kylebrowning** [comment]
@SusanDoggie what's the reasoning behind multiple configs? Is it just to swap environments?

**Comment by SusanDoggie** [comment]
it just simply allow multiple configs on the same server. It's useful when you have multiple apps connected to a server with difference configs.

**Comment by kylebrowning** [comment]
@SusanDoggie okay, I think we should change this to store multiple `APNSClient`'s. see #46 

While this doesn't relate to your case, Apple says that 
> APNs does not support authentication tokens from multiple developer accounts over a single connection.

And allowing multiple configs that get swapped on the same HTTP connection might cause issues.

**Comment by kylebrowning** [comment]
Im happy to give it a go if you dont have time to work on this right now

**Comment by kylebrowning** [comment]
closing in favor of #46

**Pull request closed without merging**